### ISSUE TITLE
Training Set have MIB group which contain different area target

### ISSUE THREAD

**Issue by KohakuBlueleaf** [IntelLabs/FloorSet#12]
Dear IntelLabs and ICCAD 2026 FloorSet organizers,

I have found that in the training set in https://huggingface.co/datasets/IntelLabs/FloorSet have tons of entry have invalid MIB groups.

By using following script to detect/reproduce the problem,
https://gist.github.com/KohakuBlueleaf/184911ab63d5a4a0a48b5a6cf7134e3a

We can find there is lot of row that have "same MIB groups id" while only the first row contain the "real MIB groups". All other row result in "the blocks that is noting in MIG group actually have different target area" which means "achiving MIB with same H,W is impossible"

We want to know if this is intended? if so, does MIB still a scoring target?

If this is not intended, will there be any updated/corrected training set?

**Comment by KohakuBlueleaf** [comment]
Some example output of that repro script:
```
python poc/repro_mib_bug.py
downloading one raw TRAIN layouts file from IntelLabs/FloorSet (streaming the tar)...
  got: floorset_lite/worker_26/layouts_5488.th

========== TRAIN raw floorset_lite/worker_26/layouts_5488.th: 112 layouts ==========
  [IMPOSSIBLE] layout 1 MIB group 1: members=[16, 38, 46, 56, 68]
               target areas=[182.0, 540.0, 540.0, 247.0, 170.0] -> 4 distinct (MIB needs all equal)
  [IMPOSSIBLE] layout 2 MIB group 1: members=[16, 38, 46, 56, 68]
               target areas=[540.0, 132.0, 420.0, 702.0, 238.0] -> 5 distinct (MIB needs all equal)
  [IMPOSSIBLE] layout 3 MIB group 1: members=[16, 38, 46, 56, 68]
               target areas=[250.0, 380.0, 600.0, 551.0, 108.0] -> 5 distinct (MIB needs all equal)
  [IMPOSSIBLE] layout 4 MIB group 1: members=[16, 38, 46, 56, 68]
               target areas=[580.0, 204.0, 42.0, 182.0, 650.0] -> 5 distinct (MIB needs all equal)
  [IMPOSSIBLE] layout 5 MIB group 1: members=[16, 38, 46, 56, 68]
               target areas=[90.0, 400.0, 81.0, 150.0, 133.0] -> 5 distinct (MIB needs all equal)
  [IMPOSSIBLE] layout 6 MIB group 1: members=[16, 38, 46, 56, 68]
               target areas=[135.0, 513.0, 378.0, 231.0, 247.0] -> 5 distinct (MIB needs all equal)
  [IMPOSSIBLE] layout 7 MIB group 1: members=[16, 38, 46, 56, 68]
               target areas=[286.0, 324.0, 77.0, 528.0, 150.0] -> 5 distinct (MIB needs all equal)
  [IMPOSSIBLE] layout 8 MIB group 1: members=[16, 38, 46, 56, 68]
               target areas=[240.0, 110.0, 299.0, 72.0, 108.0] -> 5 distinct (MIB needs all equal)
  -> 111/112 layouts have an impossible MIB group (99.1%)

========== VAL (extracted dir): 100 layouts ==========
  -> 0/100 layouts have an impossible MIB group (0.0%)
```

we can see only layout 0 have correct result and all other are not. means the MIB group member mask are duplicated wrongly to not-first row.

**Comment by KohakuBlueleaf** [comment]
NOTE the QA have mentioned the training dataset may also violate some soft constraint
BUT what I'm saying here is about "the input requirements natively contradict with soft constraint which make soft constraint is natively impossible"
